torch/distributed/checkpoint/nested_tensor.py

```diff
@@ -0,0 +1,117 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates
+
+import copy
+
+import torch.distributed as dist
+from torch.distributed.remote_device import _remote_device
+
+from torch.distributed.checkpoint.metadata import (
+    STATE_DICT_TYPE,
+)
+from torch.distributed._shard.sharded_tensor import (
+    Shard,
+    ShardMetadata,
+    ShardedTensor,
+)
+
+from torch.distributed._shard.sharded_tensor.metadata import (
+    ShardedTensorMetadata,
+)
+
+
+from .traverse import (
+    OBJ_PATH,
+    traverse_state_dict,
+    set_element,
+    STATE_DICT_ITEM,
+)
+
+from .utils import _element_wise_add
+
+
+# TODO: update docstring for nested_tensor.py
```

Comment:
Why are we creating another file with a name that's matching something existing? Is there a plan to use the actual NestedTensor here eventually? cc @drisspg